src/pymor/parameters/functionals.py

```diff
@@ -325,3 +326,134 @@ def evaluate(self, mu=None):
 
     def d_mu(self, component, index=()):
         return self.with_(constant_value=0, name=self.name + '_d_mu')
+
+
+class MinThetaParameterFunctional(ParameterFunctional):
+    """|ParameterFunctional| implementing the min-theta approach from [Haa17]_ (Proposition 2.35).
+
+    Let V denote a Hilbert space and let a: V x V -> K denote a parametric coercive bilinear form with affine
+    decomposition
+
+      a(u, v, mu) = sum_{q = 1}^Q theta_q(mu) a_q(u, v),
+
+    for Q positive coefficient |ParameterFunctional|s theta_1, ..., theta_Q and positive semi-definit component
+    bilinear forms a_1, ..., a_Q: V x V -> K. Let mu_bar be a parameter with respect to which the coercivity constant
+    of a(., ., mu_bar) is known, i.e. we known alpha_mu_bar > 0, s.t.
+
+      alpha_mu_bar |u|_V^2 <= a(u, u, mu=mu_bar).
+
+    The min-theta approach from [Haa17]_ (Proposition 2.35) allows to obtain a computable bound for the coercivity
+    constant of a(., ., mu) for arbitrary parameters mu, since
+
+      a(u, u, mu=mu) >= min_{q = 1}^Q theta_q(mu)/theta_q(mu_bar) a(u, u, mu=mu_bar).
+
+    Given a list of the thetas, the parameter mu_bar and the constant alpha_mu_bar, this functional thus evaluates
+    to ::
+
+      alpha_mu_bar * min_{q = 1}^Q theta_q(mu)/theta_q(mu_bar)
+
+
+    Parameters
+    ----------
+    thetas
+        List or tuple of |ParameterFunctional|s
+    mu_bar
+        Parameter associated with alpha_mu_bar.
+    alpha_mu_bar
+        Known coercivity constant.
+    name
+        Name of the functional.
+    """
+
+    def __init__(self, thetas, mu_bar, alpha_mu_bar=1., name=None):
+        assert isinstance(thetas, (list, tuple))
+        assert len(thetas) > 0
+        assert all([isinstance(theta, (Number, ParameterFunctional)) for theta in thetas])
+        thetas = [ConstantParameterFunctional(theta) if not isinstance(theta, ParameterFunctional) else theta
+                  for theta in thetas]
+        self.build_parameter_type(*[t for t in thetas if t.parametric])
```

Comment:
`*thetas` should work as well

Reply:
If I remember correctly, `build_parameter_type` of a list will fail if one the items is not parametric, even if it derives from ParametricInterface...